src/solve/mirk.jl

```diff
@@ -250,40 +250,40 @@ end
     resids = [get_tmp(r, u) for r in residual]
     resida = resids[1][1:prod(cache.resid_size[1])]
     residb = resids[1][(prod(cache.resid_size[1]) + 1):end]
-    eval_bc_residual!((resida, residb), pt, bc!, soly_, p, mesh)
     Φ!(resids[2:end], cache, y_, u, p)
+    eval_bc_residual!((resida, residb), pt, bc!, soly_, p, mesh)
     recursive_flatten_twopoint!(resid, resids, cache.resid_size)
     return nothing
 end
 
 @views function __mirk_loss(u, p, y, pt::StandardBVProblem, bc::BC, mesh, cache) where {BC}
     y_ = recursive_unflatten!(y, u)
-    soly_ = MIRKSol(y_, mesh, cache)
-    resid_bc = eval_bc_residual(pt, bc, soly_, p, mesh)
+    soly_ = SciMLBase.build_solution(cache.prob, cache.alg, mesh, y_, interp = MIRKInterpolation(cache.mesh, y_, cache))
```

Comment:
you should be able to build the solution type once and just keep updating its values?

Reply:
Yeah, that's better